velox/common/caching/SsdFile.cpp

```diff
@@ -309,80 +306,81 @@ std::optional<std::pair<uint64_t, int32_t>> SsdFile::getSpace(
-    auto rc = ftruncate(fd_, newSize);
+    const auto newSize = (numRegions_ + 1) * kRegionSize;
+    const auto rc = ::ftruncate(fd_, newSize);
     if (rc >= 0) {
       fileSize_ = newSize;
       writableRegions_.push_back(numRegions_);
-      regionSize_[numRegions_] = 0;
+      regionSizes_[numRegions_] = 0;
       ++numRegions_;
       return true;
-    } else {
-      ++stats_.growFileErrors;
-      LOG(ERROR) << "Failed to grow cache file " << filename_ << " to "
-                 << newSize;
     }
+
+    ++stats_.growFileErrors;
+    LOG(ERROR) << "Failed to grow cache file " << fileName_ << " to "
+               << newSize;
   }
+
   auto candidates =
       tracker_.findEvictionCandidates(3, numRegions_, regionPins_);
   if (candidates.empty()) {
     suspended_ = true;
     return false;
   }
+
   logEviction(candidates);
   clearRegionEntriesLocked(candidates);
   writableRegions_ = std::move(candidates);
   suspended_ = false;
   return true;
 }
 
-void SsdFile::clearRegionEntriesLocked(
-    const std::vector<int32_t>& regionIndices) {
+void SsdFile::clearRegionEntriesLocked(const std::vector<int32_t>& regions) {
+  std::unordered_set<int32_t> regionSet{regions.begin(), regions.end()};
   // Remove all 'entries_' where the dependent points one of 'regionIndices'.
   auto it = entries_.begin();
   while (it != entries_.end()) {
-    auto region = regionIndex(it->second.offset());
-    if (std::find(regionIndices.begin(), regionIndices.end(), region) !=
-        regionIndices.end()) {
+    const auto region = regionIndex(it->second.offset());
+    if (regionSet.count(region) != 0) {
       it = entries_.erase(it);
     } else {
       ++it;
     }
   }
-  for (auto region : regionIndices) {
-    // While the region is being filled it may get score from
-    // hits. When it is full, it will get a score boost to be a little
-    // ahead of the best.
+  for (const auto region : regions) {
+    // While the region is being filled, it may get score from hits. When it is
+    // full, it will get a score boost to be a little ahead of the best.
     tracker_.regionCleared(region);
-    regionSize_[region] = 0;
+    regionSizes_[region] = 0;
   }
 }
 
 void SsdFile::write(std::vector<CachePin>& pins) {
-  // Sorts the pins by their file/offset. In this way what is ajacent
-  // in storage is likely adjacent on SSD.
+  // Sorts the pins by their file/offset. In this way what is adjacent in
+  // storage is likely adjacent on SSD.
   std::sort(pins.begin(), pins.end());
   uint64_t total = 0;
-  for (auto& pin : pins) {
-    auto entry = pin.checkedEntry();
+  for (const auto& pin : pins) {
+    auto* entry = pin.checkedEntry();
     VELOX_CHECK_NULL(entry->ssdFile());
     total += entry->size();
   }
+
   int32_t storeIndex = 0;
   while (storeIndex < pins.size()) {
-    auto space = getSpace(pins, storeIndex);
-
-    if (!space.has_value()) {
+    auto spaceOr = getSpace(pins, storeIndex);
```

Comment:
`spaceOr` sounds incomplete. 

Other names `space`, `spaceMaybe`, `possibleSpace`, `spaceIfExists`, `optionalSpace` are better. Do we follow a convention to name optional variables? I didn't find anything.

Reply:
The absil lib has StatusOr from Google OSS: https://abseil.io/docs/cpp/guides/status#overview-of-abslstatus. We used to name the variable of StatusOr type as xxxOr. Yeah, optional is a bit different. I just revert it back.